### PR TITLE
Add Ruby 2.6 Matrix changes

### PR DESCRIPTION
**Pull request by abourque72**
This PR adds the new Matrix methods for Ruby 2.6, as well as modifies `#collect` and `#map` which now take in an optional argument. The definitions of these methods were taken from the official Ruby 2.6 implementation: https://github.com/ruby/ruby/blob/ruby_2_6/lib/matrix.rb.

### CODE REVIEW

File: lib/polyfill/v2_6/matrix.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'matrix'
+
+module Polyfill
+  module V2_6
+    module Matrix
+      def antisymmetric?
+        Matrix.Raise ErrDimensionMismatch unless square?
```
Comment (Ameji72) [inline lib/polyfill/v2_6/matrix.rb:22]: @AaronLasseigne How should this new error constant be brought in by Polyfill?

File: lib/polyfill/v2_6/matrix.rb [hunk 2]
```diff
@@ -3,6 +3,21 @@
 module Polyfill
   module V2_6
     module Matrix
+      def []=(i, j, v)
+        raise FrozenError, "can't modify frozen Matrix" if frozen?
+        (rows = check_range(i, :row)) || (row = check_int(i, :row))
```
Comment (Ameji72) [inline lib/polyfill/v2_6/matrix.rb:8]: @AaronLasseigne How should associated private methods be brought in by Polyfill?

File: lib/polyfill/v2_6/matrix.rb (hunk 1 above)
Comment (AaronLasseigne) [reply lib/polyfill/v2_6/matrix.rb:22]: In the past I've avoided bringing in anything that can't be refined into Ruby. This includes new errors. I handled this for `#delete_suffix!` (Ruby 2.5) by using the parent of the error, marking the method as partially implemented, and noting the difference.

File: lib/polyfill/v2_6/matrix.rb (hunk 2 above)
Comment (AaronLasseigne) [reply lib/polyfill/v2_6/matrix.rb:8]: I haven't had to bring in private methods yet. There are basically two options that I can think of. Either we inline the private code (which is kind of annoying) or we have a way to associate the private methods and import them along with the method requested (which is a totally new feature).

**Pull request closed without merging**